wolfcrypt/src/rsa.c

```diff
@@ -718,17 +721,27 @@ int wc_CheckRsaKey(RsaKey* key)
     }
 #endif
 
-    WC_ALLOC_VAR_EX(rng, WC_RNG, 1, NULL, DYNAMIC_TYPE_RNG,
-        return MEMORY_E);
     NEW_MP_INT_SIZE(tmp, mp_bitsused(&key->n), NULL, DYNAMIC_TYPE_RSA);
 #ifdef MP_INT_SIZE_CHECK_NULL
     if (tmp == NULL) {
-        XFREE(rng, NULL, DYNAMIC_TYPE_RNG);
         return MEMORY_E;
     }
 #endif
 
-    ret = wc_InitRng(rng);
+    if (key->rng)
```

Comment:
I don't understand need for this optimization. We're not calling wc_CheckRsaKey from linuxkm. It's used mainly in the pkcs code, and one place in wolfcrypt test.

Reply:
This was the only `wc_InitRng()` call inside the v5 FIPS boundary that was unconditional.  Everywhere else, an rng can either be set in the key context, or passed in as an arg.  Given that WolfEntropy dynamics empirically destabilize the kernel, it seemed prudent to plan ahead in case we need to be able to do PKCS#7 in kernel mode.

I've just discovered that we added an rng dependency in curve25519.c (bb84ebfd7acff) conditional on `WOLFSSL_CURVE25519_BLINDING`, and `wc_curve25519_make_pub()` has a similar issue with a local rng -- I'll add `wc_curve25519_make_pub_ex()` that adds an `rng` arg.  That's only inside the v6  FIPS boundary but it could easily come back to bite us.